vllm/transformers_utils/tokenizer_group/thread_tokenizer_group.py

```diff
@@ -0,0 +1,37 @@
+import threading
+from concurrent.futures import ThreadPoolExecutor
+
+from vllm.logger import init_logger
+from vllm.transformers_utils.tokenizer_group.tokenizer_group import (
+    TokenizerGroup)
+from vllm.utils import make_async
+
+logger = init_logger(__name__)
+
+
+class ThreadPoolTokenizerGroup(TokenizerGroup):
+    """A threadpool of TokenizerGroups for async tokenization."""
+
+    def __init__(self, *args, max_workers: int, **tokenizer_config):
+        super().__init__(*args, **tokenizer_config)
+        self.local = threading.local()
+
+        def init_tokenizer():
+            logger.info(
+                f"Starting tokenizer thread {threading.current_thread().name}")
+            self.local.tokenizer = TokenizerGroup(*args, **tokenizer_config)
+
+        self.executor = ThreadPoolExecutor(
+            max_workers=max_workers,
+            thread_name_prefix='tokenizer_thread',
+            initializer=init_tokenizer,
+        )
+
+        self.encode_async = make_async(self._encode_local, self.executor)
```

Comment:
that feels pretty hacky to me, maybe we can do sth like
```suggestion
        self._encode_async = make_async(self._encode_local, self.executor)
    
    async def encode_async(self, ...):
        return await self._encode_async(...)
```